tasks/main.yml

```diff
@@ -28,12 +28,13 @@
   service: name=jenkins state=started enabled=yes
 
 - name: Wait for Jenkins to start up before proceeding.
-  shell: "curl -D - --silent http://{{ jenkins_hostname }}:{{ jenkins_http_port }}{{ jenkins_url_prefix }}/cli/"
+  uri: url='http://{{ jenkins_hostname }}:{{ jenkins_http_port }}{{ jenkins_url_prefix }}/cli/' status_code=200
   register: result
-  until: (result.stdout.find("403 Forbidden") != -1) or (result.stdout.find("200 OK") != -1) and (result.stdout.find("Please wait while") == -1)
+  until: "(result | success) and (result.status == 200)"
```

Comment:
Technically we still need to wait until "Please wait while" is not present in the output; Jenkins responds with a 200 OK status even when it's still booting, so we can't proceed until that message is gone.